src/installing-workflow/configuring-object-storage.md

```diff
@@ -1,140 +1,117 @@
+1. Component-specific environment variables (e.g. `BUILDER_STORAGE` and `REGISTRY_STORAGE`)
+2. Access credentials stored as a Kubernetes secret named `objectstorage-keyfile`
 
-In our beta release, the components listed above are configured by default to automatically use the Minio [service][k8s-service] for object storage.
+The helm chart for Deis Workflow can be easily configured to connect Workflow components to off-cluster object storage. Deis Workflow currently supports Google Compute Storage, Amazon S3 and Azure Blob Storage.
 
-# Configuring the Deis Components
+* **Step 1:** Create buckets for each of the workflow subsystems: builder, registry and database
+* **Step 2:** Create user credentials that have access to the storage buckets from step 1
+* **Step 3:** If you haven't already fetched the helm chart, do so with `helm fetch deis/workflow-beta2`
+* **Step 4:** Open the helm chart with `helm edit workflow-beta2` and look for the template file `tpl/generate_params.toml`
+* **Step 5:** Update the `storage` parameter to reference the storage platform you are using: `s3`, `azure`, `gcs`
+* **Step 6:** Update the values in the section which corresponds to your storage type, including region, buckets and access credentials
+    * Note: you do not need to base64 encode any of these values as Helm will handle encoding automatically
+* **Step 7:** Save your changes and re-generate the helm chart by running `helm generate -x manifests workflow-beta2`
+* **Step 8:** Check the generated file in your manifests directory, you should see `deis-objectstorage-secret.yaml`
 
-Every Deis component that relies on object storage relies on the following two inputs for configuration:
+You are now ready to `helm install workflow-beta2` using your desired object storage.
 
-- An environment variable that describe what object storage system to use.
-- A configuration file ([objectstorage.toml][objectstorage-toml]) to provide access credentials for the object storage system.
-	- We suggest storing the contents of these files in [Kubernetes secrets][k8s-secret] and mounting them as volumes to each pod.
-	- See [the workflow-dev chart](https://github.com/deis/charts/tree/master/workflow-dev) for examples of using and mounting secrets.
+## Object Storage Configuration and Credentials
 
-The subsections herein explain how to configure these two inputs for each applicable component.
+During the `helm generate` step, Helm creates a Kubernetes secret in the Deis namespace named `objectstorage-keyfile`. The exact structure of the file depends on storage backend specified in `tpl/generate_params.toml`.
 
-## [deis/builder](https://github.com/deis/builder)
+```
+# Set the storage backend
+#
+# Valid values are:
+# - filesystem: Store persistent data on ephemeral disk
```

Comment:
Only registry supports the file system and it is present in the chart because initially only registry used to use this secret.So i think we should not keep it still as a valid value.Charts also need to be updated.